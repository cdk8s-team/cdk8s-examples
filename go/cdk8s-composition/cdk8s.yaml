app: go run .
language: go
imports:
  - k8s@1.26.0