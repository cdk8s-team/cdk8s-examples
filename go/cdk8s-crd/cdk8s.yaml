app: go run .
language: go
imports:
  - k8s@1.26.0
  - https://raw.githubusercontent.com/cert-manager/cert-manager/master/deploy/crds/crd-certificates.yaml