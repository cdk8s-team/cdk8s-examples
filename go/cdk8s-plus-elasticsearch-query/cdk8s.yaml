app: go run .
language: go
imports:
  - k8s@1.26.0
  - https://download.elastic.co/downloads/eck/1.1.2/all-in-one.yaml