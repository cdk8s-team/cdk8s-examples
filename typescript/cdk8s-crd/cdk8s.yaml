app: ts-node index.ts
language: typescript
imports:
  - https://raw.githubusercontent.com/cert-manager/cert-manager/master/deploy/crds/crd-certificates.yaml