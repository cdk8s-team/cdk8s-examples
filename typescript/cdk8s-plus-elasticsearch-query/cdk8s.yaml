app: ts-node index.ts
language: typescript
imports:
  - k8s@1.26.0
  - https://download.elastic.co/downloads/eck/1.1.2/all-in-one.yaml