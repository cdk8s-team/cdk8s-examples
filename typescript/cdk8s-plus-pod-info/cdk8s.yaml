app: ts-node index.ts
language: typescript
