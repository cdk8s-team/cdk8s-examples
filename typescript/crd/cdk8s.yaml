app: ts-node index.ts
language: typescript
imports:
  - k8s@1.26.0
  - https://raw.githubusercontent.com/cert-manager/cert-manager/master/deploy/crds/crd-certificates.yaml