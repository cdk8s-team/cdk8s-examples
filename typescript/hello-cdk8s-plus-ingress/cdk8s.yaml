app: ts-node index.ts
language: typescript
imports:
  - k8s@1.26.0