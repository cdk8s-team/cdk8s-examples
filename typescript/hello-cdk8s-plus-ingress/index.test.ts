import { HelloCdk8sPlusIngress } from './index';
import { Testing } from 'cdk8s';

test('snapshot', () => {
  const app = Testing.app();
  const chart = new HelloCdk8sPlusIngress(app, 'hello-cdk8s-plus-ingress');

  expect(Testing.synth(chart)).toMatchSnapshot();
});