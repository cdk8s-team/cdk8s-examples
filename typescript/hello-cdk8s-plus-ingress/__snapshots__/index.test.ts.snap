// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
[
  {
    "apiVersion": "networking.k8s.io/v1",
    "kind": "Ingress",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-ingress-c8662a2f",
    },
    "spec": {
      "rules": [
        {
          "http": {
            "paths": [
              {
                "backend": {
                  "service": {
                    "name": "hello-cdk8s-plus-ingress-root-service-c83a4af3",
                    "port": {
                      "number": 5678,
                    },
                  },
                },
                "path": "/",
                "pathType": "Prefix",
              },
              {
                "backend": {
                  "service": {
                    "name": "hello-cdk8s-plus-ingress-foo-service-c8e432fb",
                    "port": {
                      "number": 5678,
                    },
                  },
                },
                "path": "/foo",
                "pathType": "Prefix",
              },
              {
                "backend": {
                  "service": {
                    "name": "hello-cdk8s-plus-ingress-foo-bar-service-c844cebf",
                    "port": {
                      "number": 5678,
                    },
                  },
                },
                "path": "/foo/bar",
                "pathType": "Prefix",
              },
            ],
          },
        },
        {
          "host": "my.host",
          "http": {
            "paths": [
              {
                "backend": {
                  "service": {
                    "name": "hello-cdk8s-plus-ingress-my.host-hey-service-c834857c",
                    "port": {
                      "number": 5678,
                    },
                  },
                },
                "path": "/",
                "pathType": "Prefix",
              },
            ],
          },
        },
      ],
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-root-c8b2cd6a",
    },
    "spec": {
      "minReadySeconds": 0,
      "progressDeadlineSeconds": 600,
      "replicas": 2,
      "selector": {
        "matchLabels": {
          "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-root-c8d018ba",
        },
      },
      "strategy": {
        "rollingUpdate": {
          "maxSurge": "25%",
          "maxUnavailable": "25%",
        },
        "type": "RollingUpdate",
      },
      "template": {
        "metadata": {
          "labels": {
            "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-root-c8d018ba",
          },
        },
        "spec": {
          "automountServiceAccountToken": false,
          "containers": [
            {
              "args": [
                "-text",
                "root",
              ],
              "image": "hashicorp/http-echo",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": [
                {
                  "containerPort": 5678,
                },
              ],
              "resources": {
                "limits": {
                  "cpu": "1500m",
                  "memory": "2048Mi",
                },
                "requests": {
                  "cpu": "1000m",
                  "memory": "512Mi",
                },
              },
              "securityContext": {
                "allowPrivilegeEscalation": false,
                "privileged": false,
                "readOnlyRootFilesystem": true,
                "runAsNonRoot": true,
              },
              "startupProbe": {
                "failureThreshold": 3,
                "tcpSocket": {
                  "port": 5678,
                },
              },
            },
          ],
          "dnsPolicy": "ClusterFirst",
          "restartPolicy": "Always",
          "securityContext": {
            "fsGroupChangePolicy": "Always",
            "runAsNonRoot": true,
          },
          "setHostnameAsFQDN": false,
        },
      },
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-root-service-c83a4af3",
    },
    "spec": {
      "externalIPs": [],
      "ports": [
        {
          "port": 5678,
        },
      ],
      "selector": {
        "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-root-c8d018ba",
      },
      "type": "ClusterIP",
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-foo-c82d34bb",
    },
    "spec": {
      "minReadySeconds": 0,
      "progressDeadlineSeconds": 600,
      "replicas": 2,
      "selector": {
        "matchLabels": {
          "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-foo-c8e3404d",
        },
      },
      "strategy": {
        "rollingUpdate": {
          "maxSurge": "25%",
          "maxUnavailable": "25%",
        },
        "type": "RollingUpdate",
      },
      "template": {
        "metadata": {
          "labels": {
            "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-foo-c8e3404d",
          },
        },
        "spec": {
          "automountServiceAccountToken": false,
          "containers": [
            {
              "args": [
                "-text",
                "foo",
              ],
              "image": "hashicorp/http-echo",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": [
                {
                  "containerPort": 5678,
                },
              ],
              "resources": {
                "limits": {
                  "cpu": "1500m",
                  "memory": "2048Mi",
                },
                "requests": {
                  "cpu": "1000m",
                  "memory": "512Mi",
                },
              },
              "securityContext": {
                "allowPrivilegeEscalation": false,
                "privileged": false,
                "readOnlyRootFilesystem": true,
                "runAsNonRoot": true,
              },
              "startupProbe": {
                "failureThreshold": 3,
                "tcpSocket": {
                  "port": 5678,
                },
              },
            },
          ],
          "dnsPolicy": "ClusterFirst",
          "restartPolicy": "Always",
          "securityContext": {
            "fsGroupChangePolicy": "Always",
            "runAsNonRoot": true,
          },
          "setHostnameAsFQDN": false,
        },
      },
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-foo-service-c8e432fb",
    },
    "spec": {
      "externalIPs": [],
      "ports": [
        {
          "port": 5678,
        },
      ],
      "selector": {
        "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-foo-c8e3404d",
      },
      "type": "ClusterIP",
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-foo-bar-c81a0efe",
    },
    "spec": {
      "minReadySeconds": 0,
      "progressDeadlineSeconds": 600,
      "replicas": 2,
      "selector": {
        "matchLabels": {
          "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-foo-bar-c827c920",
        },
      },
      "strategy": {
        "rollingUpdate": {
          "maxSurge": "25%",
          "maxUnavailable": "25%",
        },
        "type": "RollingUpdate",
      },
      "template": {
        "metadata": {
          "labels": {
            "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-foo-bar-c827c920",
          },
        },
        "spec": {
          "automountServiceAccountToken": false,
          "containers": [
            {
              "args": [
                "-text",
                "foo-bar",
              ],
              "image": "hashicorp/http-echo",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": [
                {
                  "containerPort": 5678,
                },
              ],
              "resources": {
                "limits": {
                  "cpu": "1500m",
                  "memory": "2048Mi",
                },
                "requests": {
                  "cpu": "1000m",
                  "memory": "512Mi",
                },
              },
              "securityContext": {
                "allowPrivilegeEscalation": false,
                "privileged": false,
                "readOnlyRootFilesystem": true,
                "runAsNonRoot": true,
              },
              "startupProbe": {
                "failureThreshold": 3,
                "tcpSocket": {
                  "port": 5678,
                },
              },
            },
          ],
          "dnsPolicy": "ClusterFirst",
          "restartPolicy": "Always",
          "securityContext": {
            "fsGroupChangePolicy": "Always",
            "runAsNonRoot": true,
          },
          "setHostnameAsFQDN": false,
        },
      },
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-foo-bar-service-c844cebf",
    },
    "spec": {
      "externalIPs": [],
      "ports": [
        {
          "port": 5678,
        },
      ],
      "selector": {
        "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-foo-bar-c827c920",
      },
      "type": "ClusterIP",
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-my.host-hey-c851e454",
    },
    "spec": {
      "minReadySeconds": 0,
      "progressDeadlineSeconds": 600,
      "replicas": 2,
      "selector": {
        "matchLabels": {
          "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-my.host-hey-c872a200",
        },
      },
      "strategy": {
        "rollingUpdate": {
          "maxSurge": "25%",
          "maxUnavailable": "25%",
        },
        "type": "RollingUpdate",
      },
      "template": {
        "metadata": {
          "labels": {
            "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-my.host-hey-c872a200",
          },
        },
        "spec": {
          "automountServiceAccountToken": false,
          "containers": [
            {
              "args": [
                "-text",
                "my.host/hey",
              ],
              "image": "hashicorp/http-echo",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": [
                {
                  "containerPort": 5678,
                },
              ],
              "resources": {
                "limits": {
                  "cpu": "1500m",
                  "memory": "2048Mi",
                },
                "requests": {
                  "cpu": "1000m",
                  "memory": "512Mi",
                },
              },
              "securityContext": {
                "allowPrivilegeEscalation": false,
                "privileged": false,
                "readOnlyRootFilesystem": true,
                "runAsNonRoot": true,
              },
              "startupProbe": {
                "failureThreshold": 3,
                "tcpSocket": {
                  "port": 5678,
                },
              },
            },
          ],
          "dnsPolicy": "ClusterFirst",
          "restartPolicy": "Always",
          "securityContext": {
            "fsGroupChangePolicy": "Always",
            "runAsNonRoot": true,
          },
          "setHostnameAsFQDN": false,
        },
      },
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "name": "hello-cdk8s-plus-ingress-my.host-hey-service-c834857c",
    },
    "spec": {
      "externalIPs": [],
      "ports": [
        {
          "port": 5678,
        },
      ],
      "selector": {
        "cdk8s.io/metadata.addr": "hello-cdk8s-plus-ingress-my.host-hey-c872a200",
      },
      "type": "ClusterIP",
    },
  },
]
`;
